README.md

```diff
@@ -0,0 +1,79 @@
+# SmallRye Dux
+
+SmallRye Dux, Dux in short, is an extensible Service Discovery and Client-Side 
+Load Balancer implementation designed to work well with reactive libraries.
+Dux APIs use [SmallRye Mutiny](https://smallrye.io/smallrye-mutiny/).
+
+Dux maps service names to addresses. 
+
+Required `ServiceDiscovery` provides a list of addresses (`ServiceInstance`'s)
+for a given service name.
+Optional `LoadBalancer` selects a single `ServiceInstance` for a call.  
+
+## Configuring user applications
+The project's core aims to be vendor neutral. Aside from using Mutiny for the APIs,
+it has no dependencies. As such, it provides a universal abstraction of configuration 
+and vendors can choose their own way of providing Dux configuration.
+
+MicroProfile based frameworks can use MicroProfile Config to configure Dux. 
+Below is an example snippet of such a configuration:
+```properties
+dux.my-service.service-discovery=consul
```

Comment:
Would it be better to use a reserved config key (such as `type` or `kind`) here? Reason is that translation of this to the YAML format, which seems to be quite popular, is not exactly intuitive:

```yaml
dux:
  my-service:
    service-discovery:
      ~: consul
      some-service-discovery-property: property-value
```

MP Reactive Messaging for example uses reserved config keys, and it seems to work.

Reply:
good point, it seemed unnecessary to add a key here for properties, I didn't think of YAML.

WDYT of changing this to `dux.my-service.service-discovery.type=consul` @aureamunoz ?

Reply:
Yes, I agree with adding the key. However I don't like very much `type`, what about `tool`? 
`dux.my-service.service-discovery.tool=consul`

Reply:
it's not always a tool, it can be e.g. `static` which means a list of values provided in config by users